main/core/test/src/mill/internal/AmmoniteUtilsTests.scala

```diff
@@ -1,77 +0,0 @@
-package mill.internal
-
-import utest._
-
-object AmmoniteUtilsTests extends TestSuite {
```

Comment:
These tests reproduce important things and regressions we had after the first release (https://github.com/com-lihaoyi/mill/pull/1790 and https://github.com/com-lihaoyi/mill/pull/1822). I don't agree getting rid of them.
Also, it was kinda nice to have that pure logic in a separate file, different than `Evaluator` where a lot of things happen.
I would just rename `AmmoniteUtils` to something else `ImportUtils` and leave all the tests, so we are sure to not create regressions in the future.

Reply:
I put the tests back for now.

Reply:
Thank you @lihaoyi ❤️

Reply:
I'm wondering if there's some way to remove this logic entirely and just work with an import graph between file paths. That would save us a lot of hassle encodingg and decoding things

Reply:
That would be even better!
Especially because we pay this cost for every task evaluation, the fewer things we do, the better.